### ISSUE TITLE
long/short portfolio

### ISSUE THREAD

**Issue by Ulawun** [albertosantini/node-finance#3]
Hello Alberto,
Great work. Thanks a lot.
I'm wondering how to get the optimum for a long/short portfolio.
I checked out portfolioOptim( x, pm, lows, highs )
with the lows vector set to [ -1, -1, -1 ],
but I didn't get a solution.
Can portfolioOptim() calculate long/short portfolios ?
How to call it ?
Thanks.
Ulrich

**Comment by albertosantini** [comment]
Thanks for the feedback.

No, you can't, because I didn't need shorts in my use cases (and I should state that in the docs).

`portfolioOptim` is the JS porting of the method `portfolio.optim` in the package `tseries` in R framework.
https://cran.r-project.org/web/packages/tseries/tseries.pdf

Generally speaking, any figure is tested with an R script and only if the figures are different, it is considered a bug in node-finance.

I should add a flag `shorts` to the signature, default to `false`.

Basically I should change `lib/portfolio.js` like

```
if (shorts) {
    Amat = stats.rbind(a1, a2, a3, a3neg);
    bvec = [1].concat(pm).concat(lows).concat(highs);
} else {
    Amat = stats.rbind(a1, a2, a3, a3, a3neg);
    bvec = [1].concat(pm).concat(b3).concat(lows).concat(highs);
} 
```

Any simple test (also with R script to compare the figures)?

See `portofolio-test.js` in `test` folder.

**Comment by Ulawun** [comment]
Hello Alberto,
1000 thanks.
Your code suggestion works fantastic;
It gives my plausible results :-)
Best wishes
Ulrich
